cleverhans/model.py

```diff
@@ -46,9 +53,9 @@ def get_probs(self, x):
         :return: A symbolic representation of the output probabilities (i.e.,
                 the output values produced by the softmax layer).
         """
-        if 'probs' in self.get_layer_names():
```

Comment:
In this line, there are two motivations:
- Main motivation: don't require the `Model` subclass to implement `get_layer_names`.
- Minor motivation: use exceptions instead of tests, to remove some small runtime costs. If we use tests, then `get_layer` will query the dictionary to do the test, then query the dictionary again  to get the layer, and then we'll query the dictionary again to do the test here. So 3 dict lookups when we could have just one.